messaging/src/main/java/org/axonframework/eventhandling/async/PropertySequencingPolicy.java

```diff
@@ -0,0 +1,192 @@
+package org.axonframework.eventhandling.async;
+
+import static org.axonframework.common.BuilderUtils.assertNonNull;
+
+import java.util.function.Function;
+
+import org.axonframework.common.property.Property;
+import org.axonframework.common.property.PropertyAccessStrategy;
+import org.axonframework.eventhandling.EventMessage;
+
+/**
+ * A {@link SequencingPolicy} implementation that extracts the sequence identifier from the event message payload based
+ * on a given property or property extractor. If the event message payload is not of a supported type, a fallback 
+ * sequencing policy is used. By default the fallback sequencing policy raises an exception.
+ *
+ * @param <T>
+ *            The type of the supported event payloads.
+ * @param <K>
+ *            The type of the extracted property.
+ *
+ * @author Nils Christian Ehmke
+ * @since 4.5.2
+ */
+@SuppressWarnings( "rawtypes" )
```

Comment:
The indentation is not according to Axon's style guide. Please use the style guide as found [here](https://github.com/AxonFramework/AxonFramework/blob/master/axon_code_style.xml) and adjust the indentation accordingly.

Reply:
I formatted the code using IntelliJ now (I usually use Eclipse).